wp_api/src/login/url_discovery.rs

```diff
@@ -411,47 +433,53 @@ pub enum FetchAndParseApiRootFailure {
     },
 }
 
-impl FetchAndParseApiRootFailure {
-    pub fn error_message(&self, parsed_site_url: impl Display) -> String {
+impl WpSupportsLocalization for FetchAndParseApiRootFailure {
+    fn message_bundle(&self) -> MessageBundle {
         match self {
-            Self::FetchApiRoot { error } => error.to_string(),
-            Self::ParseApiRoot {
-                parsing_error_message,
-            } => format!("Failed to parse api details: {:#?}", parsing_error_message),
-            Self::WpError {
-                error_code,
-                error_message,
-                status_code,
-            } => format!(
-                "WpError {{\n\tstatus_code: {}\n\terror_code: {:?}\n\terror_message: \"{}\"",
-                status_code, error_code, error_message
-            ),
+            Self::FetchApiRoot { error } => error.message_bundle(),
+            Self::ParseApiRoot { .. } => WpMessages::parse_api_root(),
+            Self::WpError { error_message, .. } => WpMessages::site_error_message(error_message),
             Self::ApplicationPasswordsNotSupported { reason, .. } => reason
                 .as_ref()
-                .map(|r| r.error_message(parsed_site_url))
-                .unwrap_or("Application Passwords are not supported".to_string()),
+                .map(|r| r.message_bundle())
+                .unwrap_or(WpMessages::application_passwords_not_supported()),
         }
     }
 }
 
-#[derive(Debug, Clone, uniffi::Error)]
+#[derive(Debug, Clone, uniffi::Error, WpDeriveLocalizable)]
 pub enum ApplicationPasswordsNotSupportedReason {
     ApplicationPasswordBlockedByPlugin {
+        site_url: Arc<ParsedUrl>,
         plugin: KnownApplicationPasswordBlockingPlugin,
     },
-    ApplicationPasswordBlockedByMultiplePlugins,
+    ApplicationPasswordBlockedByMultiplePlugins {
+        site_url: Arc<ParsedUrl>,
```

Comment:
See my previous comment about moving the common fields out of sub error variants.